cpp/src/neighbors/ivf_pq/ivf_pq_search.cuh

```diff
@@ -867,9 +905,8 @@ inline void search(raft::resources const& handle,
   auto mr = raft::resource::get_workspace_resource(handle);
 
   // Maximum number of query vectors to search at the same time.
-  const auto max_queries =
-    std::min<uint32_t>(std::max<uint32_t>(n_queries, 1), params.max_internal_batch_size);
-  auto max_batch_size = get_max_batch_size(handle, k, n_probes, max_queries, max_samples);
+  const auto max_queries = get_max_queries(handle, params, n_probes, index.n_lists(), n_queries);
+  auto max_batch_size    = get_max_batch_size(handle, k, n_probes, max_queries, max_samples);
```

Comment:
Now these are batch sizes for coarse and fine search, right? We shall name them as such.